js/modules/enable-character-count.js

```diff
@@ -1,108 +1,100 @@
-    charCountInitEl = charCountEls.length > 0 ? charCountEls[0] : null;
-    const charCountTemplateEl = document.getElementById('enable-character-count__template');
-    const dataset = charCountInitEl ? charCountInitEl.dataset : {};
-
-    charCountTemplate = charCountTemplateEl ? charCountTemplateEl.innerHTML : '<span class="sr-only">${screenReaderText}</span><span aria-hidden="true">${visualText}</span>';
+    charCountEls.forEach((target) => {
+      setUpEventsFor(target);
+      setIdIfNullFor(target);
+      setUpAriaDescribedByFor(target);
+      addLiveRegion(target);
+      createCounterContainerFor(target);
+      writeCharCount(target);
+    });
+  }
 
-    globalScreenReaderTemplate = dataset.screenReaderText || 'Character Count: ${numChars} out of ${maxLength}';
-    globalVisualTemplate = dataset.visualText || '${numChars}/${maxLength}';
-    globalCounterInstructions = dataset.counterInstructions || 'Press Escape to find out how many more characters you can type.';
-    globalWarningThreshold = dataset.warningThreshold || 20;
+  function setUpEventsFor(target) {
+    target.addEventListener('keyup', onKeyUp);
+    target.addEventListener('keydown', onKeyDown);
+    target.addEventListener('focus', onFocus);
+  }
 
-    document.body.addEventListener('keyup', this.onKeyUp, true);
-    document.body.addEventListener('focus', this.onFocus, true);
+  function setIdIfNullFor(target) {
+    if (target.id == null) {
+      target.id = `enable-character-counter-${idIndex++}`;
+    }
+  }
 
-    addLiveRegion();
+  function setUpAriaDescribedByFor(target) {
+    const describedByContent = `${getScreenReaderDescription(target)} ${getScreenReaderInstructions(target)}`;
+    const ariaDescribedByElement = document.createElement('p');
+    ariaDescribedByElement.id = `${target.id}-aria-describedby`;
+    ariaDescribedByElement.className="sr-only";
+    ariaDescribedByElement.textContent = describedByContent;
+    target.insertAdjacentElement('afterend', ariaDescribedByElement);
+    target.setAttribute('aria-describedby', ariaDescribedByElement.id);
+  }
 
-    charCountEls.forEach((target, i) => {
-      setAriaDesc(target);
-      createCounterFor(target);
-      writeCharCount(target);
-    });
+  function getScreenReaderDescription(target) {
+    const { maxLength, dataset } = target;
+    const description = dataset.description ?? 'In edit text area with a ${maxLength} character limit.';
+    return interpolate(description, { maxLength });
   }
 
-  function setAriaDesc(target) {
-    const { dataset } = target;
+  function getScreenReaderInstructions(target) {
+    const { readCountKey, instructions } = target.dataset;
+    const keyToPress = readCountKey ?? defaultReadCharacterCountKey;
+    const defaultInstructions = 'Press ${keyToPress} to find out how many more characters are allowed.';
+    const instructionsToInterpolate = instructions ?? defaultInstructions;
```

Comment:
Since we don't use `defaultInstructions` anywhere else, should we just do the following?

```suggestion
    const instructionsToInterpolate = instructions ??'Press ${keyToPress} to find out how many more characters are allowed.';
```